### PR TITLE
Update cog (music player) to forked maintainted tree

### PR DESCRIPTION
**Pull request by BenjaminHCCarr**
#### Editing an existing cask
- [X] Commit message includes cask’s name (and new version, if applicable).
- [X] `brew cask audit --download cog.rb` is error-free.
- [X] `brew cask style --fix cog.rb` left no offenses.

### CODE REVIEW

**Comment by adidalal** [comment]
Any notification/official post?

Also, looks like this PR bundled a bunch of unrelated commits - it's probably better to make a new PR from a branch.

File: Casks/cog.rb [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
 
-  # the stable package on sourceforge is a bzip-inside-bzip that we can't handle
-  url "http://cogx.org/nightly_builds/cog-#{version}.tbz"
-  appcast 'http://mamburu.net/cog/stable.xml',
-          checkpoint: 'f5770d73ad0c4a19af24cf25195c01d1cc05b937a79416fe82ead0949beee62e'
+  # The abandoned cog code has been forked and is not maintained again
+  url "https://www.kode54.net/cog/mercury_builds/Cog-#{version}.zip"
   name 'Cog'
-  homepage 'http://cogx.org'
+  homepage 'https://kode54.net/cog/'
```
Comment (adidalal) [inline Casks/cog.rb]: https://gitlab.kode54.net/kode54/Cog might be better

**Comment by BenjaminHCCarr** [comment]
Yeah, I am having trouble rebasing/squashing.
I thought 'git reset --hard upstream/master' would wipe out the dead devels, but doesn't appear so.
I'll try branching and repulling.

There is no official post but the "old" repo is CR 2008, and last updated 2011. It was GPLed and mirrored a bunch of places, looks like one dev took up the cause.

File: Casks/cog.rb [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
-  version 'r2051'
-  sha256 '1cc55f718a15810cccc5141312c01079e1da230a2412e4bc66c172de7aee36fe'
+  version '4fdbf35'
+  sha256 '3239ba84b1dd65edbfa816695ffaa0af7ad6dbf6fdc3d77d9807ec2454095cbf'
 
-  # the stable package on sourceforge is a bzip-inside-bzip that we can't handle
-  url "http://cogx.org/nightly_builds/cog-#{version}.tbz"
-  appcast 'http://mamburu.net/cog/stable.xml',
-          checkpoint: 'f5770d73ad0c4a19af24cf25195c01d1cc05b937a79416fe82ead0949beee62e'
+  # The abandoned cog code has been forked and is not maintained again
```
Comment (adidalal) [inline Casks/cog.rb]: Needs a `# kode54.net/cog was verified as official when first introduced to the cask` comment above this line ([example](https://github.com/caskroom/homebrew-cask/blob/4ad02c9f087deb9e553ab0129c079d73c474add8/Casks/0ad.rb#L5); [reasoning](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment)).

File: Casks/cog.rb [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
+  url "https://www.kode54.net/cog/mercury_builds/Cog-#{version}.zip"
```
Comment (adidalal) [inline Casks/cog.rb]: use version.after_comma

File: Casks/cog.rb [hunk 4]
```diff
@@ -1,13 +1,11 @@
 cask 'cog' do
-  version 'r2051'
-  sha256 '1cc55f718a15810cccc5141312c01079e1da230a2412e4bc66c172de7aee36fe'
+  version '4fdbf35'
```
Comment (adidalal) [inline Casks/cog.rb]: Use `0.08, 4fdbf35`

**Comment by adidalal** [comment]
@BenjaminHCCarr I'll fix it

**Comment by BenjaminHCCarr** [comment]
I branched it cleanly, I can make those updates, 0.08 is the VERY old version number ~2008.
https://cogx.org/download.php 0.07 was stable 0.08 was nightly.
Looks like current dev is using some checksum, will update.

**Comment by BenjaminHCCarr** [comment]
I take it back, I didn't branch it cleanly I still have 21 commits, grrr, you step away from git for 3min and you forget everything. I was sure a rebase upstream force would fix, little did I know. Thanks for your help @adidalal

**Comment by adidalal** [comment]
https://github.com/caskroom/homebrew-cask/pull/21600

**Pull request closed without merging**